File: countrypicker/src/main/java/com/mukesh/countrypicker/BottomSheetDialogView.java
package com.mukesh.countrypicker;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.mukesh.countrypicker.listeners.BottomSheetInteractionListener;

import static com.mukesh.countrypicker.CountryPicker.THEME_NEW;

public class BottomSheetDialogView extends BottomSheetDialogFragment {
  private static final String BUNDLE_KEY_THEME = "theme";
  private BottomSheetInteractionListener listener;

  public static BottomSheetDialogView newInstance(int theme) {
    BottomSheetDialogView bottomSheetDialogView = new BottomSheetDialogView();
    Bundle args = new Bundle();
    args.putInt(BUNDLE_KEY_THEME, theme);
    bottomSheetDialogView.setArguments(args);
    return bottomSheetDialogView;
  }

  @Override public void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    Bundle args = getArguments();
    if (args != null) {
      int theme = args.getInt(BUNDLE_KEY_THEME, 0);
      if (theme == THEME_NEW) {
        setStyle(DialogFragment.STYLE_NORMAL, R.style.MaterialDialogStyle);
      } else {
        setStyle(DialogFragment.STYLE_NORMAL, R.style.DialogStyle);
      }
    } else {
      setStyle(DialogFragment.STYLE_NORMAL, R.style.DialogStyle);
    }
  }

  @Nullable @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    return inflater.inflate(R.layout.country_picker, container, false);
  }

  @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    listener.initiateUi(view);
    listener.setCustomStyle(view);
    listener.setSearchEditText();
    listener.setupRecyclerView(view);
  }

  public void setListener(BottomSheetInteractionListener listener) {
    this.listener = listener;
  }
}

File: countrypicker/src/main/java/com/mukesh/countrypicker/BottomSheetDialogView.kt
package com.mukesh.countrypicker

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.mukesh.countrypicker.listeners.BottomSheetInteractionListener

class BottomSheetDialogView : BottomSheetDialogFragment() {
    private var listener: BottomSheetInteractionListener? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val args = arguments
        if (args != null) {
            val theme = args.getInt(BUNDLE_KEY_THEME, 0)
            if (theme == CountryPicker.THEME_NEW) {
                setStyle(STYLE_NORMAL, R.style.MaterialDialogStyle)
            } else {
                setStyle(STYLE_NORMAL, R.style.DialogStyle)
            }
        } else {
            setStyle(STYLE_NORMAL, R.style.DialogStyle)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.country_picker, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        listener!!.initiateUi(view)
        listener!!.setCustomStyle(view)
        listener!!.setSearchEditText()
        listener!!.setupRecyclerView(view)
    }

    fun setListener(listener: BottomSheetInteractionListener?) {
        this.listener = listener
    }

    companion object {
        private const val BUNDLE_KEY_THEME = "theme"
        fun newInstance(theme: Int): BottomSheetDialogView {
            val bottomSheetDialogView = BottomSheetDialogView()
            val args = Bundle()
            args.putInt(BUNDLE_KEY_THEME, theme)
            bottomSheetDialogView.arguments = args
            return bottomSheetDialogView
        }
    }
}